### ISSUE TITLE
Incorrect Global Function Definition Detection for Localized Functions within do Blocks

### ISSUE THREAD

**Issue by filipovskis** [everyday-as/pivity-issues#1310]
### Expected Behavior

It should not declare localized variables as global ones.

### Actual Behavior

The code report detects an issue in actually localized variable, which it defines as a global one. The code report ignores if you prior localized the variable name. I actually do not know if it is a global issue related to localizing prior or just related to `do` blocks.

It returns an issue called: `_Global function definitions are prohibited. Namespace them inside your addon's global table_`.

## Example:
```lua
local split do
    local len = string.len
    local sub = string.sub
    local maxBytes = netchunk.maxBytes

    function split(data)
        local length = len(data)
        local index = 1
        local last = 1
        local chunks = {}

        for i = 1, length do
            if (i - last + 1) > maxBytes then
                chunks[index] = sub(data, last, i)
                index = index + 1
                last = i + 1
            end
        end

        chunks[index] = sub(data, last, length)

        return chunks
    end
end
```

## Result:
![Image](https://i.imgur.com/g5AyO37.png)

### Reproduction Steps

```markdown
1. Paste somewhere in the code the example above.
2. Run code report.
3. Check the issues.
```

### Browsers

Vivaldi

### Example URL

https://www.gmodstore.com/dashboard/products/onyx-scoreboard-a-modern-scoreboard/code-reports/rkhUPOF8QBqvkzAf40Ew7g

### Pivity Marketplace Version

v18.4.0

**Comment by Indecisiv3** [comment]
Yeah, would be amazing if this could be fixed same thing happened with one of my submissions.

**Comment by TomDotBat** [comment]
Observed this on multiple submissions, not specific to `do` blocks.

**Comment by Marcuzz** [comment]
Ah yes, we are missing a precondition here. We need to simply check whether the function definition is within another block when traversing the AST. If it is, we can safely ignore it.

I'm publishing a fix that will take effect for new versions from now on. The following are no longer considered global functions:

```lua
local split do
    function split(data)
        -- split code
    end
end
```

```lua
local function some_local_function()
    function split(data)
        -- split code
    end
end
```

Thanks for reporting! It's very useful to have actual code examples to work with.
